services/venice-server/src/main/java/com/linkedin/venice/listener/ReadQuotaEnforcementHandler.java

```diff
@@ -428,8 +448,16 @@ public void handleStoreDeleted(String storeName) {
    */
   @Override
   public void handleStoreChanged(Store store) {
+    boolean shouldLog = storeQuotaChangeMap.containsKey(store.getName())
+        && storeQuotaChangeMap.get(store.getName()) != store.getReadQuotaInCU();
```

Comment:
Nit, `get` returns the value if the key is mapped, or null if the key is not found. We can avoid a separate `containsKey` call here and we could also reuse the result later in the logging part.

Reply:
Good call, the current code is also vulnerable to races.